internal/system/snapd_client_test.go

```diff
@@ -0,0 +1,362 @@
+package system
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestSnapdSnapParsing(t *testing.T) {
```

Comment:
I think these should be tested via the exported Snap and FindOne methods, rather than directly marshalling/unmarshalling.